### DISCUSSION TITLE
ComboBoxItem data template with images?

### DISCUSSION THREAD

**Discussion by CosminCAS123** [AvaloniaUI/Avalonia#16930] [Q&A] [answered]
Hey. I want a dropdown menu with the flag of X country and its corresponding phone number Prefix. I did this :

```
<ComboBox Width="100" SelectedIndex="0" ItemsSource="{Binding PhoneNumbers}">
	<ComboBox.ItemTemplate>
		<DataTemplate>
		<StackPanel Orientation="Horizontal"
					HorizontalAlignment="Left"
					VerticalAlignment="Center" Spacing="2"
					
					>
			<Image Width="20" Height="15" Source="{Binding Uri}">
	   </Image>
			<TextBlock Text="{Binding Prefix}"></TextBlock>
		</StackPanel>
		</DataTemplate>
	</ComboBox.ItemTemplate>
	
</ComboBox>
```

But only the prefix text will appear here. Which means the bindings should work. But the image doesn't appear at all. If I manually type that URI in _Source_ in XAML it works, there is no spelling mistake.

```
 private List<PhoneNationality> phoneNationalities;
     public List<PhoneNationality> PhoneNumbers { get => this.phoneNationalities; set => this.RaiseAndSetIfChanged(ref this.phoneNationalities , value); }

     public SecondRegisterVM()
     {
     this.PhoneNumbers = new List<PhoneNationality>
           {
                new PhoneNationality(PhoneResources.MaxDigits.France, PhoneResources.Prefixes.France, PhoneResources.URIs.France),
           };
     }
```

(simplified version of the ViewModel). And PhoneNationality is just a class with 3 properties(maxdigits,prefix, and URI), which I also made reactive thinking that it would make the difference, and it didn't.  What can be the issue here?

**Comment by IanRawley** [comment]
I'm guessing the URI property of PhoneNationality is a string?
If so, based on the [documentation](https://docs.avaloniaui.net/docs/reference/controls/image), it probably either needs to be pointing to an Avalonia resource, or you need to use a converter to load and return an actual image.  I'm also not sure binding to a string representing an avares: URI would actually work, as the documentation does say "a string constant naming an application asset".

**Reply by CosminCAS123** [reply]
Yes, it is a string. Thank you.

**Comment by stevemonaco** [comment, marked as answer]
This works with a constant string (not binded) from XAML because XAML has some magic involving type converters. See:

https://github.com/AvaloniaUI/Avalonia/blob/77fe0400fc57139c4a53fc0c7d148fca36c80cdd/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AvaloniaXamlIlLanguage.cs#L180-L182

When binding, you need to: 1. bind to an `IImage` (not appropriate for a VM) since that's what `Image.Source` expects or 2. write a converter (not good for images that require async loading because converters and async don't mix well).

**Reply by IanRawley** [reply]
An Async Image Loading Behaviour would probably also work, although would still need to be written yourself.  A little more verbose in XAML, but gets around the Async converter problem.
